### PR TITLE
20/03/enh/quotation optimizations2

### PR DESCRIPTION
**Pull request by melanore**

### CODE REVIEW

**Comment by kerams** [comment]
Ooooh, this looks tasty.

>rowmapping inferred in runtime

Wanted to tackle this too some time in the future. Great.

File: paket.dependencies [hunk 1]
```diff
@@ -0,0 +1,32 @@
+storage: packages
+source https://nuget.org/api/v2
```
Comment (kerams) [inline paket.dependencies]: You might consider changing this to `https://api.nuget.org/v3/index.json` (same in group Test) for better perf.

Comment (melanore) [reply paket.dependencies]: @kerams, review appreciated. There is some additional work required, but core functionality is there. Also, for some reason I can not restore test db any more, - it fails with "pg_restore: [archiver] unsupported version (1.14) in file header" error. Update of postgres didn't help...

Comment (kerams) [reply paket.dependencies]: I made the backup on 12.1 using pgAdmin. That could be the problem.

**Comment by kerams** [comment]
I'm getting this after updating:
```
error NU1202: Package FSharp.Data.Npgsql 0.2.0-beta is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package FSharp.Data.Npgsql 0.2.0-beta supports:
- net461 (.NETFramework,Version=v4.6.1)
- netstandard2.0 (.NETStandard,Version=v2.0)
```

**Comment by melanore** [comment]
There are some problems with cross targeting, for now I will separate switch to new template version from performance optimizations. Pushed a new version with old template. Sorry.

**Comment by kerams** [comment]
No worries. I have another issue now though :). Building inside VS works, but `dotnet build` fails with

```
error FS3031: The type provider 'C:\Users\_\.nuget\packages\fsharp.data.npgsql\0.2.2-beta\lib\netstandard2.0\FSharp.Data.Npgsql.dll' reported an error:
Assembly attribute 'TypeProviderAssemblyAttribute' refers to a designer assembly 'FSharp.Data.Npgsql.DesignTime' which cannot be loaded or doesn't exist.
Could not load file or assembly 'C:\Users\_\.nuget\packages\fsharp.data.npgsql\0.2.2-beta\lib\netstandard2.0\FSharp.Data.Npgsql.DesignTime.dll'
```

**Pull request merged**